### PR TITLE
feat(over window): general streaming over window frontend part

### LINKED ISSUE #9932: General version of streaming over window function, incl. `lag` and `lead`
The syntax is identical with the one stated here: https://github.com/risingwavelabs/risingwave/issues/8962

### LINKED ISSUE #10072: frontend: `StreamOverWindow` node (general version)

### PR DESCRIPTION
**Pull request by stdrc**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Resolves #10072 and resolves #9932. Depends on #10624.

- Add `StreamOverWindow` plan node
- Connect from `LogicalOverWindow` to `OverWindowExecutor`
- Re-arrange planner tests
- Add e2e tests

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- ~~[ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- ~~[ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)~~

## Documentation

- [x] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/frontend/planner_test/tests/testdata/output/over_window_function.yaml [hunk 1]
```diff
@@ -44,15 +101,47 @@
...
+    create table t(x int, y int);
+    select x, y, lag(x) over(PARTITION BY y ORDER BY x) from t;
+  logical_plan: |-
+    LogicalProject { exprs: [t.x, t.y, first_value] }
+    └─LogicalOverWindow { window_functions: [first_value(t.x) OVER(PARTITION BY t.y ORDER BY t.x ASC ROWS BETWEEN 1 PRECEDING AND 1 PRECEDING)] }
+      └─LogicalProject { exprs: [t.x, t.y, t._row_id] }
+        └─LogicalScan { table: t, columns: [t.x, t.y, t._row_id] }
+  stream_plan: |-
+    StreamMaterialize { columns: [x, y, t._row_id(hidden), lag], stream_key: [t._row_id], pk_columns: [t._row_id], pk_conflict: NoCheck }
+    └─StreamExchange { dist: HashShard(t._row_id) }
```
Comment (BugenZhao) [inline src/frontend/planner_test/tests/testdata/output/over_window_function.yaml:115]: Should we have the stream key also include the partition key `y` here?

**Review by BugenZhao — COMMENTED** [review]
Generally LGTM

File: src/frontend/planner_test/tests/testdata/output/over_window_function.yaml (hunk 1 above)
Comment (stdrc) [reply src/frontend/planner_test/tests/testdata/output/over_window_function.yaml]: Yes, it's fixed

File: src/stream/src/executor/over_window/general.rs [hunk 2]
```diff
@@ -372,16 +372,17 @@ impl<S: StateStore> OverWindowExecutor<S> {
...
-                        // split into delete + insert, will be merged after building changes
                         part_delta.insert(old_state_key, Change::Delete);
                         part_delta.insert(new_state_key, Change::Insert(new_row.into_owned_row()));
                     } else {
-                        // partition-change update
-                        key_change_updated_pks.insert(this.get_input_pk(old_row));
-                        // split into delete + insert, will be merged after building changes
+                        // partition-change update, split into delete + insert
+                        // NOTE(rc): Since we append partition key to logical pk, we can't merge the
+                        // delete + insert back to update later.
```
Comment (BugenZhao) [inline src/stream/src/executor/over_window/general.rs:384]: Good catch.

**Review by BugenZhao — APPROVED** [review]
LGTM

**Review by st1page — APPROVED** [review]
LGTM

**Pull request merged**